### PR TITLE
Recipe for SDL2 and pysdl2

### PR DESCRIPTION
**Pull request by dschreij**
Conda-build recipes for compiling SDL2, and its wrapper pysdl2, for all platforms.

### CODE REVIEW

File: recipes/pysdl2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+{% set version = "0.9.4" %}
+{% set md5 = "720b33ed048c169643406ae39ca4dd8d" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn:  {{ name }}-{{ version }}.tar.gz
+  url:  https://pypi.python.org/packages/2d/e3/05d37f4d83c87f59ccd3f3e65bd5a9df200fa03c9f6472ac0fd8fcce23db/{{ name }}-{{ version }}.tar.gz
```
Comment (patricksnape) [inline recipes/pysdl2/meta.yaml]: `url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz`

File: recipes/pysdl2/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  license: ZLIB
+  summary: "A wrapper around the SDL2 library"
+  description: |
+    PySDL2 is a wrapper around the SDL2 library and as such similar to the discontinued PySDL project. In contrast to PySDL, it has no licensing restrictions, nor does it rely on C code, but uses ctypes instead.
+  doc_url: https://pysdl2.readthedocs.io
+  dev_url: https://bitbucket.org/marcusva/py-sdl2
+
+extra:
+  recipe-maintainers:
+    - dschreij
```
Comment (patricksnape) [inline recipes/pysdl2/meta.yaml]: Missing terminal new line

File: recipes/pysdl2/meta.yaml (hunk 1 above)
Comment (patricksnape) [inline recipes/pysdl2/meta.yaml]: `url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz`

File: recipes/pysdl2/meta.yaml (hunk 2 above)
Comment (patricksnape) [inline recipes/pysdl2/meta.yaml]: Missing terminal new line

File: recipes/pysdl2/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+test:
+  imports:
+    - sdl2
+
+about:
+  home: http://bitbucket.org/marcusva/py-sdl2
+  license: ZLIB
+  summary: "A wrapper around the SDL2 library"
+  description: |
+    PySDL2 is a wrapper around the SDL2 library and as such similar to the discontinued PySDL project. In contrast to PySDL, it has no licensing restrictions, nor does it rely on C code, but uses ctypes instead.
```
Comment (patricksnape) [inline recipes/pysdl2/meta.yaml]: Can we truncate this to < 100 characters per line.

File: recipes/sdl2/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn:  {{ name }}-{{ version }}.tar.gz  # [unix]
+  url:  https://www.libsdl.org/release/{{ name }}-{{ version }}.tar.gz  # [unix]
+  sha256: {{ sha256 }}  # [unix]
+  fn:  {{ name }}-{{ version }}.zip  # [win]
+  url:  https://www.libsdl.org/release/{{ name }}-{{ version }}.zip  # [win]
```
Comment (patricksnape) [inline recipes/sdl2/meta.yaml]: sha256 for this windows zip file?

File: recipes/sdl2/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+build:
+  number: 0
+  features:
+    - vc9  # [win and py27]
+    - vc10  # [win and py34]
+    - vc14  # [win and py35]
+  script_env:
+    - LOCALAPPDATA  # [win and py27]
```
Comment (patricksnape) [inline recipes/sdl2/meta.yaml]: Why do we need this?

File: recipes/sdl2/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  build:
+    - cmake  # [win]
+    - autoconf  # [unix]
+
+about:
+  home: https://www.libsdl.org/index.php
+  license: ZLIB
+  summary: "Low-level access to audio, keyboard, mouse, joystick, and graphics"
+  description: |
+    Simple DirectMedia Layer is a cross-platform development library designed to provide low level access to audio, keyboard, mouse, joystick, and graphics hardware via OpenGL and Direct3D.
```
Comment (patricksnape) [inline recipes/sdl2/meta.yaml]: Can we truncate this?

File: recipes/sdl2/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  doc_url: https://wiki.libsdl.org/FrontPage
+  dev_url: http://hg.libsdl.org/SDL
+
+extra:
+  recipe-maintainers:
+    - dschreij
```
Comment (patricksnape) [inline recipes/sdl2/meta.yaml]: Terminal new line

File: recipes/sdl2/build.sh [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+
+ # we have to do this because most build scripts assume that all sdl modules
+ # are installed to the same prefix.
+sed -i -- "s|@prefix@|${PREFIX}|g" sdl2.pc.in 
+sed -i -- "s|@prefix@|${PREFIX}|g" sdl2-config.in
+
+# Build SDL2
+./autogen.sh
+./configure --prefix=${PREFIX} --disable-haptic --without-x
+make install
```
Comment (patricksnape) [inline recipes/sdl2/build.sh]: Terminal new line

Comment (patricksnape) [inline recipes/sdl2/build.sh]: Also, is there a `make check`?

File: recipes/sdl2/bld.bat [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+cd build
+
+call %LIBRARY_BIN%\cmake -G "NMake Makefiles" -DCMAKE_INSTALL_PREFIX:PATH="%LIBRARY_PREFIX%" -DCMAKE_BUILD_TYPE:STRING=!TARGET! !DIRECTX_FLAG! .. 
+if errorlevel 1 exit 1
+
+nmake
+if errorlevel 1 exit 1
+
+nmake install
+if errorlevel 1 exit 1
```
Comment (patricksnape) [inline recipes/sdl2/bld.bat]: Terminal new line

File: recipes/sdl2/bld.bat [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+	set DIRECTX_FLAG="-DDIRECTX=OFF"
+)
+if errorlevel 1 exit 1
+
+:: Go to the source folder
+cd %SRC_DIR%
+mkdir build
+cd build
+
+call %LIBRARY_BIN%\cmake -G "NMake Makefiles" -DCMAKE_INSTALL_PREFIX:PATH="%LIBRARY_PREFIX%" -DCMAKE_BUILD_TYPE:STRING=!TARGET! !DIRECTX_FLAG! ..
```
Comment (patricksnape) [inline recipes/sdl2/bld.bat]: Do we need the `call`?

File: recipes/sdl2/bld.bat [hunk 11]
```diff
@@ -0,0 +1,40 @@
+:: MSVC library for Python 2.7 is not found automaticaly if it is 
+:: installed in the user's AppData folder. Add it to PATH
+setlocal EnableDelayedExpansion
+
+set TARGET=Release
+:: Set target back to Debug for 64-bits builds with VS2008 and VS2010
+if %VS_MAJOR% LSS 14 (
+	if %ARCH% == 64 (
+		set TARGET=Debug
```
Comment (patricksnape) [inline recipes/sdl2/bld.bat]: Why?

File: recipes/sdl2/bld.bat [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+if %VS_MAJOR% LSS 14 (
+	if %ARCH% == 64 (
+		set TARGET=Debug
+	)
+)
+
+:: When installing Visual C++ compiler tools for Python 2.7, all files are placed in the users AppData folder
+:: conda-bld tries to call vcvarsall at its classic location, which of course fails, so we have to call it from its
+:: new location in the the users appdata folder manually, but only if this folder exists of course.
+if "%PY_VER%"=="2.7" (
```
Comment (patricksnape) [inline recipes/sdl2/bld.bat]: Is `conda-build` not behaving here? Do we need this for Appveyor?

File: recipes/sdl2/bld.bat [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+		if %ARCH% == 32 set VC_ARCH=x86
+		if %ARCH% == 64 set VC_ARCH=amd64
+		call !VC9DIR!\vcvarsall.bat !VC_ARCH!
+	)
+	set DIRECTX_FLAG="-DDIRECTX=OFF"
+)
+if errorlevel 1 exit 1
+
+:: Go to the source folder
+cd %SRC_DIR%
```
Comment (patricksnape) [inline recipes/sdl2/bld.bat]: Do we need this?

File: recipes/sdl2/build.sh [hunk 14]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+cd ${SRC_DIR}
```
Comment (patricksnape) [inline recipes/sdl2/build.sh]: Do we need this?

File: recipes/pysdl2/meta.yaml (hunk 3 above)
Comment (dschreij) [inline recipes/pysdl2/meta.yaml]: Yes, will do.

File: recipes/pysdl2/meta.yaml (hunk 2 above)
Comment (dschreij) [inline recipes/pysdl2/meta.yaml]: Will add it

File: recipes/sdl2/meta.yaml (hunk 5 above)
Comment (dschreij) [inline recipes/sdl2/meta.yaml]: To be able to find the VC++ tools for Python in bld.bat.

File: recipes/sdl2/meta.yaml (hunk 6 above)
Comment (dschreij) [inline recipes/sdl2/meta.yaml]: Yes

File: recipes/sdl2/meta.yaml (hunk 7 above)
Comment (dschreij) [inline recipes/sdl2/meta.yaml]: Will add it

File: recipes/sdl2/build.sh (hunk 8 above)
Comment (dschreij) [inline recipes/sdl2/build.sh]: Will add it

File: recipes/sdl2/meta.yaml (hunk 4 above)
Comment (dschreij) [inline recipes/sdl2/meta.yaml]: Had to generate that myself, but totally forgot about it. Will add the sha256 (08ba1de8bb4236d3aa1a94168b6e1a4f286bd21ca3437c11689d4e55f3a9cc89) to the recipe.

File: recipes/sdl2/build.sh (hunk 8 above)
Comment (dschreij) [inline recipes/sdl2/build.sh]: Not that I know of. I largely based the build process on the one created for Homebrew (OS X)

File: recipes/sdl2/bld.bat (hunk 11 above)
Comment (dschreij) [inline recipes/sdl2/bld.bat]: Otherwise there is a linker error referencing the function memset, which doesn't occur when the Target is anything else than release, also see https://github.com/conda/conda-build/issues/1374

File: recipes/sdl2/bld.bat (hunk 12 above)
Comment (dschreij) [inline recipes/sdl2/bld.bat]: Yes, this is also necessary for appveyor, as the VC tools are also installed in a user folder. If you look at the appveyor log of this package when built for Python 2.7 (https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.7155/job/h1invbjev7vrekql) you'll see:

~~~
(root) C:\Miniconda-x64\conda-bld\work\SDL2-2.0.4>call "C:\Program Files (x86)\Microsoft Visual Studio 9.0\Common7\Tools\..\..\VC\vcvarsall.bat" amd64 
The specified configuration type is missing.  The tools for the
configuration might not be installed.
~~~

around line 350 indicating that vcvarsall cannot be found at this location. Later at line 361 you see

~~~
-- Check for working C compiler: C:/Users/appveyor/AppData/Local/Programs/Common/Microsoft/Visual C++ for Python/9.0/VC/bin/amd64/cl.exe
-- Check for working C compiler: C:/Users/appveyor/AppData/Local/Programs/Common/Microsoft/Visual C++ for Python/9.0/VC/bin/amd64/cl.exe -- works
~~~
which means that, also on appveyor, the VC tools are found in the users appdata folder. conda-build does not check this location by default, hence I've added this snippet to my build scripts which guards for this.

File: recipes/sdl2/bld.bat (hunk 13 above)
Comment (dschreij) [inline recipes/sdl2/bld.bat]: I don't know, does the build script run from the source directory by default? This is placed with the idea of being "better safe than sorry".

File: recipes/sdl2/bld.bat (hunk 10 above)
Comment (dschreij) [inline recipes/sdl2/bld.bat]: No we don't! I forgot this is only necessary when calling other bat files from within a bat file. I will remove it.

File: recipes/sdl2/bld.bat (hunk 9 above)
Comment (dschreij) [inline recipes/sdl2/bld.bat]: I will add it.

File: recipes/sdl2/build.sh (hunk 14 above)
Comment (dschreij) [inline recipes/sdl2/build.sh]: See the comment with the similar issue at the bld.bat file. Shall I remove it?

File: recipes/sdl2/bld.bat (hunk 13 above)
Comment (patricksnape) [inline recipes/sdl2/bld.bat]: Yes it runs in source directory by default.

File: recipes/sdl2/build.sh (hunk 14 above)
Comment (patricksnape) [inline recipes/sdl2/build.sh]: Yes

File: recipes/sdl2/bld.bat (hunk 12 above)
Comment (patricksnape) [inline recipes/sdl2/bld.bat]: [OpenCV also fails](https://ci.appveyor.com/project/conda-forge/opencv-feedstock/build/1.0.63/job/7f0tnchs8ejbbttt#L286) to grab the `amd64` compiler due the configuration files being missing with the vanilla MSVC 2008. However, this doesn't matter as CMake still picks up the correct compiler for NMake files. Are you sure it doesn't work without all this logic?

File: recipes/sdl2/bld.bat (hunk 11 above)
Comment (patricksnape) [inline recipes/sdl2/bld.bat]: That `memset` error smells like the release build removing the CRT intrinsics to me. May have something to do with all this business about not being reliant on any CRT version.

File: recipes/pysdl2/meta.yaml [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn:  {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  md5: {{ md5 }}
+  patches:
+    - locate_sdl.patch
```
Comment (jakirkham) [inline recipes/pysdl2/meta.yaml]: Has this patch been submitted upstream? If not, could you please submit it as a PR? Please xref the PR used for this patch in a comment above it in the recipe.

File: recipes/pysdl2/meta.yaml [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+
+build:
+  number: 0
+  script: python setup.py install
```
Comment (jakirkham) [inline recipes/pysdl2/meta.yaml]: Please add ` --single-version-externally-managed --record record.txt` when using `setuptools` to build.

File: recipes/pysdl2/meta.yaml [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+    - sdl2
+    - python
+
+test:
+  imports:
+    - sdl2
+
+about:
+  home: http://bitbucket.org/marcusva/py-sdl2
+  license: ZLIB
```
Comment (jakirkham) [inline recipes/pysdl2/meta.yaml]: So this is Public Domain where permissible. Would update this to `Public Domain/ZLIB`.

Comment (jakirkham) [inline recipes/pysdl2/meta.yaml]: Also please add the license file like so.

```yaml
  license_file: doc/copying.rst
```

File: recipes/sdl2/meta.yaml [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+
+build:
+  number: 0
+  features:
+    - vc9  # [win and py27]
+    - vc10  # [win and py34]
+    - vc14  # [win and py35]
+
+requirements:
+  build:
```
Comment (jakirkham) [inline recipes/sdl2/meta.yaml]: Please add `    - python  # [win]`. Otherwise we won't get the proper builds we expect.

File: recipes/sdl2/meta.yaml [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+    - cmake  # [win]
+    - autoconf  # [unix]
+
```
Comment (jakirkham) [inline recipes/sdl2/meta.yaml]: Would recommend adding the existing test here.

```yaml
test:
  commands:
    - sdl2-config --version
```

File: recipes/sdl2/meta.yaml [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+about:
+  home: https://www.libsdl.org/index.php
+  license: ZLIB
```
Comment (jakirkham) [inline recipes/sdl2/meta.yaml]: Please add the license file like so.

```yaml
  license_file: COPYING.txt
```

File: recipes/sdl2/meta.yaml (hunk 19 above)
Comment (jakirkham) [inline recipes/sdl2/meta.yaml]: Also may be worth adding some existence tests for header files and libraries.

File: recipes/sdl2/run_test.sh [hunk 21]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+${PREFIX}/bin/sdl2-config --version
```
Comment (jakirkham) [inline recipes/sdl2/run_test.sh]: Would drop this file and add the command to the test section mentioned above. This should give you Windows testing for free. 😄

File: recipes/pysdl2/meta.yaml (hunk 15 above)
Comment (dschreij) [inline recipes/pysdl2/meta.yaml]: Thanks for reviewing! I'm not exactly sure what you mean. I included this patch from the beginning in the main recipe. I cannot place the patch in the recipe itself and have to add it as a separate PR? I'm a bit confused about what I have to do now. I have to remove the patch here, commit, push, and then include the patch and send it as a separate pull request?

File: recipes/pysdl2/meta.yaml (hunk 16 above)
Comment (dschreij) [inline recipes/pysdl2/meta.yaml]: Will do

File: recipes/pysdl2/meta.yaml (hunk 17 above)
Comment (dschreij) [inline recipes/pysdl2/meta.yaml]: Will do

File: recipes/sdl2/meta.yaml (hunk 18 above)
Comment (dschreij) [inline recipes/sdl2/meta.yaml]: Will do

File: recipes/sdl2/meta.yaml (hunk 19 above)
Comment (dschreij) [inline recipes/sdl2/meta.yaml]: The only thing is that sdl-config is not available in Windows as an executable, so that won't work there. I can add this and then add the # [unix] selector?

File: recipes/sdl2/run_test.sh (hunk 21 above)
Comment (dschreij) [inline recipes/sdl2/run_test.sh]: True, but afaik sdl-config only works on unix systems. The Windows builds don't come with something like that (at least, the executable is not built and installed when building with CMake)

File: recipes/sdl2/meta.yaml (hunk 19 above)
Comment (dschreij) [inline recipes/sdl2/meta.yaml]: Additionally, since I can't run any kind of test for Windows, should I add the python requirement with the windows selector?

File: recipes/sdl2/meta.yaml (hunk 20 above)
Comment (dschreij) [inline recipes/sdl2/meta.yaml]: Done (Before you proposed doc/copying.rst, but I chose for this one)

File: recipes/sdl2/bld.bat (hunk 12 above)
Comment (dschreij) [inline recipes/sdl2/bld.bat]: I removed this section and indeed it worked flawlessly. I remembered hitting my head over it when building QScintilla2, and that wouldn't build without this section, but apparently conda-build solves this problem on its own now.

File: recipes/sdl2/bld.bat (hunk 11 above)
Comment (dschreij) [inline recipes/sdl2/bld.bat]: I am not really a C expert so I have no idea what CRT intrinsics are ;). One solution would be to add a patch for sdl2 itself, but I think I lack the knowledge to do this, so I hope the current solution suffices.

File: recipes/sdl2/bld.bat (hunk 12 above)
Comment (patricksnape) [inline recipes/sdl2/bld.bat]: Yes I believe I fixed this upstream but I wanted to make sure before I investigated it again 😄

File: recipes/pysdl2/meta.yaml (hunk 15 above)
Comment (patricksnape) [inline recipes/pysdl2/meta.yaml]: Keep the patch here - but propose this patch as a PR on the upstream SDL branch so that other can benefit from it/we have less chance of the patch failing from future upstream changes.

File: recipes/sdl2/meta.yaml [hunk 22]
```diff
@@ -0,0 +1,51 @@
...
+
+requirements:
+  build:
+    - python  # [win]
+    - cmake  # [win]
+    - autoconf  # [unix]
+
+test:
+  requires:
+    - python {{ environ['PY_VER'] + '.*' }}  # [win]
```
Comment (patricksnape) [inline recipes/sdl2/meta.yaml]: I don't believe you need this if you aren't running any tests on Windows.

File: recipes/pysdl2/meta.yaml (hunk 16 above)
Comment (dschreij) [inline recipes/pysdl2/meta.yaml]: This breaks the builds on all platforms, because it doesn't know the `--single-version-externally-managed ` flag:
~~~
C:\Miniconda35-x64\conda-bld\work\PySDL2-0.9.4>call activate.bat C:\Miniconda35-x64\envs\_build 
usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help
error: option --single-version-externally-managed not recognized
Command failed: cmd.exe /c bld.bat
Command exited with code 1
~~~
(example from appveyor, but Travis and Circle crash on the same)

File: recipes/sdl2/meta.yaml (hunk 22 above)
Comment (dschreij) [inline recipes/sdl2/meta.yaml]: I removed it.

**Comment by dschreij** [comment]
Anything new on this one I should take care of?

File: recipes/sdl2/meta.yaml [hunk 23]
```diff
@@ -0,0 +1,49 @@
...
+    - autoconf  # [unix]
+
+test:
+  commands:
+    - sdl2-config --version  # [unix]
+
+about:
+  home: https://www.libsdl.org/index.php
+  license: Public Domain/ZLIB
+  license_file: COPYING.txt
```
Comment (patricksnape) [inline recipes/sdl2/meta.yaml]: This path is relative to the source directory, so the `COPYING.txt` you are providing are not being used. Therefore, you can remove them.

File: recipes/pysdl2/meta.yaml (hunk 15 above)
Comment (jakirkham) [inline recipes/pysdl2/meta.yaml]: Has this patch been proposed upstream yet? It need not be merged already, but we do like to have our patch numbers decreasing if at all possible.

File: recipes/pysdl2/meta.yaml (hunk 2 above)
Comment (jakirkham) [inline recipes/pysdl2/meta.yaml]: note: Done.

File: recipes/sdl2/build.sh (hunk 8 above)
Comment (jakirkham) [inline recipes/sdl2/build.sh]: Note: Done.

Comment (jakirkham) [inline recipes/sdl2/build.sh]: Sorry, this is a bit unclear. Is there a `check`, `test` or other testing step in the `Makefile`?

File: recipes/sdl2/meta.yaml [hunk 24]
```diff
@@ -0,0 +1,49 @@
...
+    - vc10  # [win and py34]
+    - vc14  # [win and py35]
+
+requirements:
+  build:
+    - python  # [win]
+    - cmake  # [win]
+    - autoconf  # [unix]
+
+test:
```
Comment (jakirkham) [inline recipes/sdl2/meta.yaml]: Please add the following.

```yaml
  requires:
    - python {{ environ['PY_VER'] + '*' }}  # [win]
```

**Comment by jakirkham** [comment]
Seems like there are a couple of things to take care of @dschreij . Any chance you will have time to look?

**Comment by dschreij** [comment]
Yes, I have some spare time to dive into this again the next few days, so I will!

**Comment by jakirkham** [comment]
Sorry think I closed this by accident.

File: recipes/sdl2/meta.yaml (hunk 23 above)
Comment (dschreij) [inline recipes/sdl2/meta.yaml]: Removed COPYING.txt from recipe folder

**Comment by dschreij** [comment]
Sorry for the commit history mess again. No idea how to do this cleanly when rebasing with the staged-recipes master branch and my own work...

**Comment by dschreij** [comment]
the circleci build failed, because the package built for pysdl2 could not be found. It appears though as if the pysdl2 package was built and installed with python3.5, while it was intended to be used with python2.7. Additionally, the required sdl2 package, did not appear in the list of to be installed packaged in the test environment. Don't know how much I can do about this if my meta.yaml is (assumably) correct?

**Comment by dschreij** [comment]
Alright, everything builds correctly again and your comments have been processed. Is there anything else that needs work in this recipe? If not, I'm going to continue with sdl2_gfx, ttf, mixer, net and image, which will be quite a challenge for Windows by the looks of it.

**Comment by jakirkham** [comment]
Some of the CIs failed. So am trying to restart them.

**Comment by dschreij** [comment]
Thanks @jakirkham, but I see they failed again. The reason is that I run pysdl2's offical test suite in the test phase, and there lies the predicament. For one, it tries to test some graphical drawing functions and apparently on some of the CI's there is no mock X-Server (or the like) available, causing these tests to fail. Secondly, pysdl2 tries to invocate the add-on libraries sdl2_gfx, sdl2_image and sdl2_ttf which are not available by default (I'm working on these right now), and which are optional; (py)sdl2 can perfectly be used without them. My question now is, should I abandon the test suite because it is too specific and depends on a graphical environment which is not available on some platforms? Without the elaborate tests of pysdl2, the packages build correctly.

File: recipes/sdl2/meta.yaml [hunk 25]
```diff
@@ -0,0 +1,51 @@
...
+build:
+  number: 0
+  features:
+    - vc9  # [win and py27]
+    - vc10  # [win and py34]
+    - vc14  # [win and py35]
+
+requirements:
+  build:
+    - python {{ environ['PY_VER'] + '*' }}  # [win]
```
Comment (jakirkham) [inline recipes/sdl2/meta.yaml]: No need for this jinja here. It is only needed in the `test/requires` section.

File: recipes/pysdl2/meta.yaml [hunk 26]
```diff
@@ -0,0 +1,51 @@
...
+
+build:
+  number: 0
+  script: python setup.py install --record record.txt
+
+requirements:
+  build:
+    - patch  # [win]
+    - sdl2
+    - python {{ environ['PY_VER'] + '*' }}
```
Comment (jakirkham) [inline recipes/pysdl2/meta.yaml]: No need for this jinja here.

**Comment by jakirkham** [comment]
Sounds like testing is a pain. I'd make an issue in the feedstock afterwards. Let's focus on cleaning these up so they are mergeable. Sound good?

**Comment by jakirkham** [comment]
To clarify, simply comment the tests that are causing issues.

**Comment by jakirkham** [comment]
So we like to keep each PR limited to one recipe or perhaps a couple related recipes. However, this is a few too many. Could we please break them up into smaller PRs? 😄

**Comment by dschreij** [comment]
Haha sure. I was afraid that you guys would find this a bit too much. 

The problem is that this is all one big dependency tree (except for the image and audio decoding libraries, which can be built well on their own). All sdl2_* packages depend on sdl2 and some in turn depend on decoding libraries. I don't know how to otherwise make PRs for recipes that don't (yet) have their dependencies on conda-forge. I'm going to break it down in smaller pieces now nevertheless, and we'll just see where we end up.

**Comment by dschreij** [comment]
I'm closing this PR and starting over...
